test/unit/rules/require-each-key-test.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const { ERROR_MESSAGE } = require('../../../lib/rules/require-each-key');
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'require-each-key',
+
+  config: true,
+
+  good: [
+    '{{#each this.items key="id" as |item|}} {{item.name}} {{/each}}',
+    '{{#each this.items key="deeply.nested.id" as |item|}} {{item.name}} {{/each}}',
+    '{{#each this.items key="@index" as |item|}} {{item.name}} {{/each}}',
+    '{{#each this.items key="@identity" as |item|}} {{item.name}} {{/each}}',
```

Comment:
Need to make sure that this rule only applies to `each` by adding this as a valid case:

```
{{#if foo}}{{/if}}
```

Reply:
done